File: src/main/java/com/arika/ArchiveDatabase/models/PhysicalAsset.java
package com.arika.ArchiveDatabase.models;

import java.util.ArrayList;
import java.util.List;


// an asset is a physical item in the archive. Not a digital archive item. Those come under documents.
public class PhysicalAsset {

    private String type;

    private String format;

    private String description;

    private String itemNumber;

    private String notes;

    private String location;

    private List<Document> documents;

    private List<Artist> artists;

    private List<SingleEvent> singleEvents;

    public PhysicalAsset(String type, String format, String description, String itemNumber, String notes, String location) {
        this.type = type;
        this.format = format;
        this.description = description;
        this.itemNumber = itemNumber;
        this.notes = notes;
        this.location = location;
        this.documents = new ArrayList<Document>();
        this.artists = new ArrayList<Artist>();
        this.singleEvents = new ArrayList<SingleEvent>();

    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getFormat() {
        return format;
    }

    public void setFormat(String format) {
        this.format = format;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getItemNumber() {
        return itemNumber;
    }

    public void setItemNumber(String itemNumber) {
        this.itemNumber = itemNumber;
    }

    public String getNotes() {
        return notes;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public List<Document> getDocuments() {
        return documents;
    }

    public void setDocuments(List<Document> documents) {
        this.documents = documents;
    }

    public List<Artist> getArtists() {
        return artists;
    }

    public void setArtists(List<Artist> artists) {
        this.artists = artists;
    }

    public List<SingleEvent> getSingleEvents() {
        return singleEvents;
    }

    public void setSingleEvents(List<SingleEvent> singleEvents) {
        this.singleEvents = singleEvents;
    }
}

File: frontend/archive_database/src/components/physicalAssets/PhysicalAsset.js
import React from "react";

const PhysicalAsset = (props) => {

  if(!props.physicalAsset){
    return null;
  }

  // const id = props.physicalAsset.id

  const type = props.physicalAsset.type
  const format = props.physicalAsset.format
  const description = props.physicalAsset.description
  const itemNumber = props.physicalAsset.itemNumber
  const notes = props.physicalAsset.notes
  const location = props.physicalAsset.location


  const documents = props.physicalAsset.documents.map((digitalAsset, index)=>{
    return(
      <li key = {digitalAsset.id}>
      <div>
      {digitalAsset.id}
      <div>
      {digitalAsset.fileName}
      </div>
      </div>
      </li>
    )
  })

  const artists = props.physicalAsset.artists.map((artist, index)=>{
    return (
      <li key={artist.id}>
      <div>
      {artist.id}
      <div>
      <li>{artist.name}</li>
        <li></li>
      </div>
      </div>
      </li>
    )
  })

return(
  <div className = "single-physicalAsset">
  <div className="details">

    <ul>
    <li>Single Event Title: {props.physicalAsset.singleEvent.title}</li>
    <li>Type: {props.physicalAsset.type}</li>
    <li>Format: {props.physicalAsset.format}</li>
    <li>Description: {props.physicalAsset.description}</li>
    <li>Item Number: {props.physicalAsset.itemNumber}</li>
    <li>location:{props.physicalAsset.location} </li>
    <li>Notes:{props.physicalAsset.notes} </li>
    <li>Documents associated with this item:</li>
    {documents}
    <li>Artists associated with this item:</li>
    {artists}
    </ul>
  </div>
  </div>
)


}

export default PhysicalAsset;
